core/frontend/src/PerModelCategoryVisibility.ts

```diff
@@ -103,27 +112,77 @@ class PerModelCategoryVisibilityOverrides extends SortedArray<PerModelCategoryVi
       return PerModelCategoryVisibility.Override.None;
   }
 
-  public setOverride(modelIds: Id64Arg, categoryIds: Id64Arg, override: PerModelCategoryVisibility.Override): void {
+  /**
+   *
+   * @param perModelCategoryVisibility array of model category visibility overrides
+   * @param viewState Optional param viewState. If no viewState is provided, then the viewState associated with the viewport (used to construct this class) is used. This param is WIP, not sure if necessary.
+   * Messaged OCP team to figure this one out. Waiting on response.
+   * @returns void promise
+   */
+  public async setOverrides(perModelCategoryVisibility: PerModelCategoryVisibilityProps[], viewState?: ViewState): Promise<void> {
+    let anyChanged = false;
+    const addedCategoryIds: string[] = [];
+    const viewStateToUse = viewState ? viewState : this._vp.view;
+    for (const override of perModelCategoryVisibility) {
+      const modelId = override.modelId;
+      const categoryIds = override.categoryIds;
+      const visOverride = override.visOverride;
+      for (const categoryId of Id64.iterable(categoryIds)) {
+        if (this.findAndUpdateOverrideInArray(modelId, categoryId, visOverride)) {
+          anyChanged = true;
+          this._vp.setViewedCategoriesPerModelChanged();
```

Comment:
Slightly unsure.. The reason I did it was because in the original setOverride its called twice as well, but I see thats slightly different. Changed it